internal/headers/headers.go

```diff
@@ -11,15 +10,14 @@ import (
 	"github.com/rs/zerolog"
 )
 
-func basicAuth(username, password string) string {
-	auth := username + ":" + password
-	return base64.StdEncoding.EncodeToString([]byte(auth))
-}
-
-func addIdentityHeader(ctx context.Context, req *http.Request, username, password string) error {
-	if username != "" && password != "" {
-		zerolog.Ctx(ctx).Warn().Msgf("Username/password authentication: %s", username)
-		req.Header.Add("Authorization", "Basic "+basicAuth(username, password))
+func addIdentityHeader(ctx context.Context, req *http.Request, issuerUrl, clientID, clientSecret string) error {
+	if clientID != "" && clientSecret != "" {
+		token, err := getToken(ctx, issuerUrl, clientID, clientSecret)
+		if err != nil {
+			return err
+		}
+		zerolog.Ctx(ctx).Warn().Msgf("Service account authentication: %s", clientID)
```

Comment:
This should not be warning, maybe info or debug.

Reply:
This is warning, because we don't want that to happen in Stage/Prod, so this makes it easier to spot in case by some accident we allow it there.

Reply:
Or that was the thinking why it was a warning for basic auth, I've kept the level.